test/profiles.js

```diff
@@ -0,0 +1,111 @@
+import 'babel-polyfill'
+import { assert } from 'chai'
+import fs from 'fs'
+
+import {
+  _setupBaseAndTabularRegistryMocks,
+  fetchMock
+} from './_mocks'
+
+import Util from '../src/utils'
+import Profiles from '../src/profiles'
+
+describe('Profiles', () => {
+  const DEFAULT_REGISTRY_URL = 'http://schemas.datapackages.org/registry.csv'
+
+  afterEach(() => {
+    fetchMock.restore()
+  })
+
+  describe('#retrieve', () => {
+    beforeEach(() => {
+      _setupBaseAndTabularRegistryMocks(DEFAULT_REGISTRY_URL)
+    })
+
+    it('returns undefined if profile ID doesn\'t exist', async () => {
+      const profiles = await new Profiles(true)
+        , retrieved = profiles.retrieve('inexistent-profile-id')
+
+      assert(retrieved === undefined, 'Value should be undefined')
+    })
+
+    it('returns remote profile by its ID', async () => {
```

Comment:
Please rewrite it without mocking - we better should be sure it works against a real remote schemas instead of saving a few seconds for tests.